components/src/core/channel-value/channel-value.component.ts

```diff
@@ -15,9 +16,28 @@ import { requireInput } from '../../utils/utils';
             <span class="pxb-channel-value-icon-wrapper">
                 <ng-content></ng-content>
             </span>
-            <div *ngIf="units && prefix" class="pxb-channel-value-units">{{ units }}</div>
-            <div *ngIf="value" class="pxb-channel-value-value">{{ value }}</div>
-            <div *ngIf="units && !prefix" class="pxb-channel-value-units">{{ units }}</div>
+            <div
+                *ngIf="units && prefix"
+                class="pxb-channel-value-units"
+                [class.pxb-channel-value-units-prefix]="!checkWhiteListUnits() || unitSpace === 'show'"
+            >
+                {{ units }}
+            </div>
+            <div
+                *ngIf="value"
+                class="pxb-channel-value-value"
+                [class.pxb-channel-value-add-space]="unitSpace === 'auto' || unitSpace === 'show'"
+                [class.pxb-channel-value-remove-space]="unitSpace === 'hide'"
+            >
+                {{ value }}
+            </div>
+            <div
+                *ngIf="units && !prefix"
+                class="pxb-channel-value-units"
+                [class.pxb-channel-value-units-suffix]="!checkWhiteListUnits() || unitSpace === 'show'"
```

Comment:
Instead of using an inline-function here, could you move introduce a variable `isWhiteListedUnited` that is updated `onChanges`.  This way the fn won't run each CD cycle.

Reply:
Ideally `checkWhiteListUnits` will never have to run outside of `ngOnChanges`.

Reply:
I have removed `checkWhiteListUnits`. Please review once.